internal/services/containers/kubernetes_cluster_resource_test.go

```diff
@@ -227,6 +228,22 @@ func TestAccKubernetesCluster_updateNetworkProfileOutboundType(t *testing.T) {
 			Config: r.networkProfileWithOutboundType(data, "loadBalancer"),
 		},
 		data.ImportStep(),
+		// 'none' tested in network isolated setup with bootstrap profile
+	})
+}
+
+func TestAccKubernetesCluster_bootstrapProfile(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_kubernetes_cluster", "test")
+	r := KubernetesClusterResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.networkIsolatedBootstrapProfile(data, managedclusters.ArtifactSourceCache),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("bootstrap_profile.0.artifact_source").HasValue("Cache"),
+			),
+		},
```

Comment:
After every test step, we like to add the `data.ImportStep()` to confirm there is no diff from what is in state to what is returned by Azure. Adding this step also removes the need for the `HasValue` check.

```suggestion
			),
		},
		data.ImportStep(),
```